File: minimum_working/Get_B_L.m
function B_L = Get_B_L(alpha,L,NumOfVertexs,Sigma,D_Tau,lambda,TempSlice,K,T_hop,Miu,Uene)
    %Diag = eye([NumOfVertexs,NumOfVertexs])*(Miu-Uene/2);
    Sigma_L = Sigma(L,:);
    V_L = zeros([NumOfVertexs,NumOfVertexs]);
    for i = 1:1:NumOfVertexs
       V_L(i,i) = exp(lambda*alpha*Sigma_L(i));
    end %Buide Up Matrix V_L
     B_L = V_L*expm(D_Tau*(K*T_hop));
%     if L == 0 || L == TempSlice
%         B_L = V_L;
%     end
end

File: Armadillo/Get_B_L.cpp
#include "DQMC.h"
mat Get_B_L(int alpha, int L, int NumOfVertexs, mat Sigma,double D_Tau, double lambda, int TempSlice, mat K, double T_hop, double Miu, double Uene)
{
	mat B_L(NumOfVertexs, NumOfVertexs);
	mat Sigma_L = Sigma.row(L);
	mat V_L(NumOfVertexs, NumOfVertexs, fill::zeros);
	for (int i = 0; i < NumOfVertexs; i++)
	{
		V_L(i, i) = exp(lambda * alpha * Sigma_L[i]);
	};
	B_L = V_L * expmat(D_Tau * T_hop * K);
	return B_L;
}